components/Ripple/src/MaterialState.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ Enumerates all Material States.
+
+ The integert value used for each State matches the UIControlState value so both states
+ maybe used interchangeably. For example:
+
+ [rippleView setRippleColor:UIColor.redColor forState:(MaterialState)UIControlStateHighlighted];
+
+ @note Additional Material states that do not have a corresponding UIControlState value are
+      within UIControlState's range that is reserved for application use.
+ */
+typedef NS_OPTIONS(NSInteger, MaterialState) {
```

Comment:
The PR summary states that this can be used in APIs that accept UIControlState values. Does this work for Swift as well? 

If we are introducing new values for UIControlState API arguments, I think it's better for us to actually provide UIControlState values.  That would mean also providing an appropriate bitmask and performing any recommended internal announcements/discussions about our intention to take control of those values within Google.  I think we should be *very cautious* about introducing these values and allowing their use on UIControl subclasses (like MDCButton or MDCChip). 

My biggest concern is that a client may be using the same Application-reserved range for other purposes on an MDC component and that our reassignment of the values can cause unexpected side-effects.  I think it's important to note that `UIControlStateApplication` is named *Application* and not *User* or something else. Given the number of levels of downstream frameworks that consume our components before they get to an application, there's a great opportunity for conflicts.

Reply:
Agreed with the sentiment here and the encouragement of caution.

I believe the risk of overlapping is somewhat mitigated because we own the components for which these new control states apply. I'd expect overlap to exist if someone were to subclass one of our components and add new states, but I expect this is a rarity.

Reply:
Spoke with Rob offline a bit more about his concerns, summarizing them here for visibility.

In essence this proposal is reserving certain bits of UIControlState for MDC-specific use. This will not affect all UIControlState use in an app, but it could potentially impact anyone that is subclassing our components and adding new UIControlState values. As it stands, many of our stateful components can be subclassed, so it will be healthy for us to reach out to the community to evaluate the impact of reserving some UIControlState bits at the MDC level. Notably, I'm hopeful the community will share any subclasses that may have reserved their own UIControlState bits so that we can define a migration path or influence which bits we decide to reserve for MDC.

In essence, we'll want to start a request for comment for the following proposal: "we'd like to reserve the following UIControlState bits (defined bits here) for all MDC classes; this will affect you if you subclass any of these components and add your own states, and we'd like to better understand the impact of this proposal"